pkg/apps/util/util.go

```diff
@@ -782,6 +788,23 @@ func DeploymentsForCleanup(configuration *deployapi.DeploymentConfig, deployment
 	return relevantDeployments
 }
 
+func ConfigExceededTimeoutSeconds(config *deployapi.DeploymentConfig, latestRC *v1.ReplicationController) bool {
+	var timeoutSeconds int64
+	if params := config.Spec.Strategy.RollingParams; params != nil {
+		timeoutSeconds = deployapi.DefaultRollingTimeoutSeconds
+		if params.TimeoutSeconds != nil {
```

Comment:
I think this is the wrong property check the description here https://github.com/openshift/origin/blob/fb1679e6acc6882bf25599890d1b23664a8d6548/pkg/apps/apis/apps/types.go#L272-L273
Seems like activeDeadlineSeconds is what we use for timeout https://github.com/openshift/origin/blob/fb1679e6acc6882bf25599890d1b23664a8d6548/pkg/apps/apis/apps/types.go#L219-L221

Reply:
also general timeout shouldn't be dependent on strategy

Reply:
I think we want  // TimeoutSeconds is the time to wait for updates before giving up. If the 
 // value is nil, a default will be used. 

Because literally this thing is about not getting any progress since we are not able to create deployer pod. ActiveDeadlineSeconds sets the duration for the deployer pod, but in this case there is no deployer pod.